src/index.ts

```diff
@@ -403,37 +400,39 @@ export class Upload extends Pumpify {
     }
 
     if (this.userProject) {
-      reqOpts.params = reqOpts.params || {};
-      reqOpts.params.userProject = this.userProject;
+      reqOpts.qs = reqOpts.qs || {};
+      reqOpts.qs.userProject = this.userProject;
     }
 
-    reqOpts.validateStatus = (status: number) => {
-      return (status >= 200 && status < 300) ||
-          status === RESUMABLE_INCOMPLETE_STATUS_CODE;
-    };
-
-    this.authClient.request(reqOpts).then(
-        r => {
-          return callback(null, r, r.data);
-        },
-        (err: AxiosError) => {
-          const body = err.response ? err.response.data : undefined;
-          const e = (body && body.error) ? body.error : err;
-          return callback(e, err.response, body);
+    this.authClient.getRequestHeaders()
+        .then(authHeaders => {
+          reqOpts.headers = Object.assign({}, reqOpts.headers, authHeaders);
+          request(reqOpts, (err, res, body) => {
+            const e = (body && body.error) ? body.error : err;
+            if ((res.statusCode >= 200 && res.statusCode < 300) ||
```

Comment:
@stephenplusplus @kinwa91 this is making my head hurt. I know this code doesn't make sense, but all of the unit and system tests are passing.

What's this supposed to do different in the event of a !2xx?

Reply:
I don't think it's applicable to `request` unless we use `request-promise`.

From axios' docs:

```
  // `validateStatus` defines whether to resolve or reject the promise for a given
  // HTTP response status code. If `validateStatus` returns `true` (or is set to `null`
  // or `undefined`), the promise will be resolved; otherwise, the promise will be
  // rejected.
  validateStatus: function (status) {
    return status >= 200 && status < 300; // default
  },
```

Reply:
On closer look, it added `RESUMABLE_INCOMPLETE_STATUS_CODE` (HTTP code 308) to be considered to have been successful, thereby resolves the promise.

Reply:
Here we probably need to call `callback(null, res, body);` for the when `code >= 200 < 300 || RESUMABLE_INCOMPLETE_STATUS_CODE`?